### PR TITLE
Authenticate with OmniAuth

### PR DESCRIPTION
**Pull request by Rinbo**
PT: https://www.pivotaltracker.com/story/show/161039594

- Add new controller for OmniAuth callbacks
- Add OmniAuth required migrations
- Modify user model with omniauth response methods
- Add test fixture that sends back facebook data including the Affiliation tag
- Setup login account on Facebook and add keys to encrypted credentials

### CODE REVIEW

**Comment by tochman** [comment]
Is there a feature in your PT for this?

File: app/views/devise/shared/_links.html.erb [hunk 1]
```diff
@@ -20,6 +20,6 @@
 
 <%- if devise_mapping.omniauthable? %>
   <%- resource_class.omniauth_providers.each do |provider| %>
-    <%= link_to "Sign in with #{OmniAuth::Utils.camelize(provider)}", omniauth_authorize_path(resource_name, provider) %><br />
+    <%= link_to "Sign in with #{OmniAuth::Utils.camelize(provider)}", user_facebook_omniauth_authorize_path(resource_name, provider) %><br />
```
Comment (tochman) [inline app/views/devise/shared/_links.html.erb]: I'm not sure about iterating over the providers AND hardcoding the facebook oauth path...

**Review by tochman — COMMENTED** [review]
Looks good, but I left a comment

**Comment by Rinbo** [comment]
> Is there a feature in your PT for this?

Yes, I have updated the description of the PR with the link to PT.

**Review by tochman — APPROVED** [review]
LGTM

**Pull request merged**